root/static/scripts/common/components/EntityLink.js

```diff
@@ -41,9 +41,9 @@ const Comment = ({className, comment}: {|+className: string, +comment: string|})
   </>
 );
 
-const EventDisambiguation = ({event}: {|+event: EventT|}) => {
+const EventDisambiguation = ({event, showDate}: {|+event: EventT, +showDate: boolean|}) => {
   const dates = formatDatePeriod(event);
-  if (!dates && !event.cancelled) {
+  if (showDate === false || (!dates && !event.cancelled)) {
```

Comment:
That'll cause it to return null even if `event.cancelled` is true. The check should be `((!dates || !showDate) && !event.cancelled)`.

Then you need to use `showDate` to exclude the date below:

```JSX
{dates && showDate ? ' ' + bracketed(dates) : null}
```